xonsh/environ.py

```diff
@@ -1971,7 +1971,9 @@ def get(self, key, default=None):
             return default
 
     def __iter__(self):
-        yield from (set(self._d) | set(self._vars))
+        for key in set(self._d) | set(self._vars):
+            if isinstance(key, str):
+                yield key
```

Comment:
This should maybe return the pattern itself (a string) if this is a regex.  As it stands, I think this will break some of the docs

Reply:
How can it break the docs? Returning a string representation of the regex does not make any sense, because you can not address such a key by string.

Reply:
We generate docs for the regular expressions keys.  See `docs/conf.py` down at the bottom.  Now these keys will be skipped I believe

Reply:
OK, I see. `conf.make_envvars()` iterates over `xonsh.environ.DEFAULT_VARS`, so it's not affected by this change.
If you think that it have sense to return the patterns, I can open a PR for cbd0001 that allows access to the RE variables.

Reply:
I think maybe we could have a separate iterator which returns all the keys.  I am fine with the default `__iter__` method returning just strings.